platform/ontology/api.go

```diff
@@ -45,99 +47,165 @@ func (p *Platform) GetTokenTxsByAddress(address string, token string) (blockatla
+			if err == nil {
+				txsChan <- txRaw
 			}
-			tx.Date = int64(txn.TxnTime)
-			tx.Coin = coin.Ontology().ID
-			txs = append(txs, *tx)
+		}(blockTxRaw, &wg)
+	}
+	wg.Wait()
+	close(txsChan)
+	if len(txsChan) != len(txnList) {
+		return nil, errors.E("getTxDetails failed to call client.GetTxDetailsByHash http get or unmarshal")
+	}
+	var txsRaw []TxV2
+	for tx := range txsChan {
+		if len(tx.TxDetails.Transfers) > 0 {
+			txsRaw = append(txsRaw, tx)
 		}
-		block.Txs = txs
 	}
-
-	return &block, nil
+	return txsRaw, nil
 }
 
-func Normalize(srcTx *Tx, assetName string) (tx blockatlas.Tx, ok bool) {
-	if len(srcTx.TransferList) < 1 {
-		return tx, false
+func normalizeBlock(blockOnt BlockResult, txsRaw []TxV2) (block *blockatlas.Block) {
+	block = &blockatlas.Block{
+		Number: int64(blockOnt.Result.Height),
+		ID:     blockOnt.Result.Hash,
 	}
-	transfer := srcTx.TransferList[0]
-	fee := numbers.DecimalExp(srcTx.Fee, 9)
-	var status blockatlas.Status
-	if srcTx.ConfirmFlag == 1 {
-		status = blockatlas.StatusCompleted
-	} else {
-		status = blockatlas.StatusFailed
+	block.Txs = normalizeBlockTransactions(txsRaw)
+	return
+}
+
+func normalizeBlockTransactions(txsRaw []TxV2) (txs []blockatlas.Tx) {
+	if len(txsRaw) > 0 {
+		for _, txRaw := range txsRaw {
+			tx := normalizeBlockTransaction(txRaw)
+			txs = append(txs, *tx)
+		}
 	}
+	return
+}
 
-	tx = blockatlas.Tx{
-		ID:     srcTx.TxnHash,
+func normalizeBlockTransaction(txRes TxV2) *blockatlas.Tx {
+	fee := numbers.DecimalExp(txRes.Fee, ONGDecimalsAmount)
+	tx := blockatlas.Tx{
+		ID:     txRes.TxHash,
 		Coin:   coin.ONT,
+		Date:   int64(txRes.TxTime),
+		Block:  uint64(txRes.BlockHeight),
+		Status: calculateTxStatus(txRes.ConfirmFlag),
 		Fee:    blockatlas.Amount(fee),
-		Date:   srcTx.TxnTime,
-		Block:  srcTx.Height,
-		Status: status,
 	}
 
-	switch assetName {
-	case ONTAssetName:
-		normalizeONT(&tx, &transfer)
-	case ONGAssetName:
-		normalizeONG(&tx, &transfer)
-	default: // unsupported asset
-		return tx, false
+	if len(txRes.TxDetails.Transfers) > 0 {
+		txDetails := txRes.TxDetails.Transfers[0]
+		switch txDetails.AssetName {
+		case ONTAssetName:
+			normalizeONT(&tx, Transfer{
+				Amount:      txDetails.Amount,
+				FromAddress: txDetails.FromAddress,
+				ToAddress:   txDetails.ToAddress,
+			})
+		case ONGAssetName:
+			normalizeONG(&tx, Transfer{
+				Amount:      txDetails.Amount,
+				FromAddress: txDetails.FromAddress,
+				ToAddress:   txDetails.ToAddress,
+			})
+		default: // unsupported asset
+			return &tx
+		}
 	}
+	return &tx
+}
 
-	return tx, true
+func calculateTxStatus(confirmFlag int) (status blockatlas.Status) {
+	if confirmFlag == 1 {
+		status = blockatlas.StatusCompleted
+	} else {
+		status = blockatlas.StatusFailed
+	}
+	return
 }
 
-func normalizeONT(tx *blockatlas.Tx, transfer *Transfer) {
+func normalizeONT(tx *blockatlas.Tx, transfer Transfer) {
 	i := strings.IndexRune(transfer.Amount, '.')
-	value := transfer.Amount[:i]
+	var value string
```

Comment:
Use https://github.com/trustwallet/blockatlas/blob/master/pkg/numbers/amount.go to parse numbers

Reply:
there is no suitable operation there